tests/model_serving/model_runtime/triton/basic_model_deployment/conftest.py

```diff
@@ -0,0 +1,248 @@
+from typing import cast, Any, Generator, List, Dict
+import copy
+import pytest
+from contextlib import contextmanager
+
+from kubernetes.dynamic.exceptions import ResourceNotFoundError
+from syrupy.extensions.json import JSONSnapshotExtension
+from pytest_testconfig import config as py_config
+
+from kubernetes.dynamic import DynamicClient
+from ocp_resources.namespace import Namespace
+from ocp_resources.serving_runtime import ServingRuntime
+from ocp_resources.inference_service import InferenceService
+from ocp_resources.pod import Pod
+from ocp_resources.secret import Secret
+from ocp_resources.template import Template
+from ocp_resources.service_account import ServiceAccount
+
+from tests.model_serving.model_runtime.triton.constant import (
+    PREDICT_RESOURCES,
+    RUNTIME_MAP,
+)
+from tests.model_serving.model_runtime.triton.basic_model_deployment.utils import (
+    get_template_name,
+    get_gpu_identifier,
+)
+
+from utilities.constants import (
+    KServeDeploymentType,
+    Protocols,
+)
+from utilities.inference_utils import create_isvc
+from utilities.infra import get_pods_by_isvc_label
+from utilities.serving_runtime import ServingRuntimeFromTemplate
+
+from simple_logger.logger import get_logger
+
+LOGGER = get_logger(name=__name__)
+
+
+@pytest.fixture(scope="class")
+def triton_grpc_serving_runtime_template(
+    admin_client: DynamicClient, triton_runtime_image: str
+) -> Generator[Template, None, None]:
+    with create_triton_template(
+        admin_client=admin_client, protocol=Protocols.GRPC, triton_runtime_image=triton_runtime_image
+    ) as template:
+        yield template
+
+
+@pytest.fixture(scope="class")
+def triton_rest_serving_runtime_template(
+    admin_client: DynamicClient, triton_runtime_image: str
+) -> Generator[Template, None, None]:
+    with create_triton_template(
+        admin_client=admin_client, protocol=Protocols.REST, triton_runtime_image=triton_runtime_image
+    ) as template:
+        yield template
+
+
+@contextmanager
+def create_triton_template(
+    admin_client: DynamicClient, protocol: str, triton_runtime_image: str
+) -> Generator[Template, Any, Any]:
+    template_dict = {
+        "apiVersion": "template.openshift.io/v1",
+        "kind": "Template",
+        "metadata": {
+            "name": f"triton-{protocol}-runtime-template",
+            "namespace": py_config["applications_namespace"],
+        },
+        "objects": [create_triton_serving_runtime(protocol=protocol, triton_runtime_image=triton_runtime_image)],
+        "parameters": [],
+    }
```

Comment:
for kind dict, you don't need to pass the kind and apiVersion, if I am not mistaken. pass the namespace to create_triton_serving_runtime() call here so that you can get the objects